### PR TITLE
[Feature 7B] Creating Matches

### LINKED ISSUE #10: [Feature 7B] Creating Matches
When two users both like each other, this should be detected by the system. The users should then be able to see a default generated chat window to inform them that a match has been made and prompt them to start connecting with each other.

### PR DESCRIPTION
**Pull request by rsrinivasan1**
Whenever the ```currentUser``` clicks on the displayed user, the ```RecommendationView``` calls upon the ```RecommendationPresenter``` to trigger the ```checkForMatchAndCreate``` method in the ```MatchInteractor``` class. If a match is created, we display a success popup and add the displayed user to the contact list.

Fixes #10

Added methods relating to the recommendation feed use case to maintain clean architecture.

# Checklist
- [x] My PR does not contain unrelated files that may cause conflicts
- [x] I have performed a self-review of my own code
- [x] I have commented my code with javadoc and perhaps in-line comments (particularly in
  hard-to-understand areas)
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have not violated clean architecture principles
- [x] I have opted to use the most general form of datastructures (example: use `List` instead of
  `ArrayList` wherever possible)

### CODE REVIEW

File: app/src/androidTest/java/com/group80/uoftinder/MatchInteractorUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+package com.group80.uoftinder;
```
Comment (helen-li) [inline app/src/androidTest/java/com/group80/uoftinder/MatchInteractorUnitTest.java:1]: Quick question did you end up moving the original UpdateListTest tests over here?

File: app/src/main/java/com/group80/uoftinder/MatchInteractor.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     * If user2 has already liked currentUser, then we add the two users to each other's
+     * match list, re-upload the users to the database, and send an introductory message on
+     * behalf of currentUser
+     *
+     * Preconditions
+     *   - currentUser.getLiked().contains(user2.getUid());
+     * @param currentUser is the user currently using the app who clicked like
+     * @param user2 is the user whose liked list we need to check and possibly create a match with
+     * @return a boolean indicating whether a match has been created
+     */
```
Comment (helen-li) [inline app/src/main/java/com/group80/uoftinder/MatchInteractor.java:33]: Nice documentation here!

**Review by nicholaspoon03 — APPROVED** [review]
Good work, Raghav! Looks good

File: app/src/main/java/com/group80/uoftinder/MatchInteractor.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            UserRealtimeDbFacade.uploadUser(currentUser);
+
+            sendIntroMessage(currentUser.getUid(),
+                    user2.getUid(), "Hey! We matched with each other!"); // send a message from currentUser to user2
+            return true;
+        }
+        return false;
+    }
+
+    private static void sendIntroMessage(String selfUid, String contactUid, String message) {
```
Comment (helen-li) [inline app/src/main/java/com/group80/uoftinder/MatchInteractor.java:51]: Javadoc here would be nice

**Review by helen-li — COMMENTED** [review]
Looks good overall, Raghav! Just dropped some quick comments :)

File: app/src/androidTest/java/com/group80/uoftinder/MatchInteractorUnitTest.java (hunk 1 above)
Comment (nicholaspoon03) [reply app/src/androidTest/java/com/group80/uoftinder/MatchInteractorUnitTest.java:1]: Yes, that is correct

**Review by helen-li — APPROVED** [review]
Looks great, thanks for the quick turnaround Raghav!

File: app/src/main/java/com/group80/uoftinder/MatchInteractor.java (hunk 2 above)
Comment (rsrinivasan1) [reply app/src/main/java/com/group80/uoftinder/MatchInteractor.java:33]: Fixed!

File: app/src/main/java/com/group80/uoftinder/MatchInteractor.java (hunk 3 above)
Comment (rsrinivasan1) [reply app/src/main/java/com/group80/uoftinder/MatchInteractor.java:58]: Fixed!

**Review by nicholaspoon03 — APPROVED** [review]

**Review by LanceLi1416 — APPROVED** [review]

**Pull request merged**